packages/static-hosting/lib/static-hosting.ts

```diff
@@ -1,144 +1,99 @@
-import { Construct, CfnOutput, RemovalPolicy } from "@aws-cdk/core";
+import { Construct } from "constructs";
+import { CfnOutput, Duration, RemovalPolicy } from "aws-cdk-lib";
+import { Certificate } from "aws-cdk-lib/aws-certificatemanager";
 import {
-  Bucket,
-  BucketEncryption,
-  BlockPublicAccess,
-  BucketProps,
-} from "@aws-cdk/aws-s3";
-import {
-  OriginAccessIdentity,
-  CloudFrontWebDistribution,
+  Distribution,
+  DistributionProps,
+  HttpVersion,
   PriceClass,
-  ViewerProtocolPolicy,
+  ResponseHeadersPolicy,
   SecurityPolicyProtocol,
   SSLMethod,
-  Behavior,
-  SourceConfiguration,
-  CloudFrontWebDistributionProps,
-  LambdaEdgeEventType,
+  ViewerProtocolPolicy,
+  BehaviorOptions,
+  ErrorResponse,
+  EdgeLambda,
   CfnDistribution,
-  ResponseHeadersPolicy,
-  HttpVersion,
-} from "@aws-cdk/aws-cloudfront";
-import { HostedZone, ARecord } from "@aws-cdk/aws-route53";
-import { User, Group, Policy, PolicyStatement, Effect } from "@aws-cdk/aws-iam";
-import { Version } from "@aws-cdk/aws-lambda";
-import { ArbitraryPathRemapFunction } from "./arbitrary-path-remap";
+  OriginRequestPolicy,
+  CachePolicy,
+  OriginRequestHeaderBehavior,
+  CacheHeaderBehavior,
+  IResponseHeadersPolicy,
+} from "aws-cdk-lib/aws-cloudfront";
+import { S3Origin } from "aws-cdk-lib/aws-cloudfront-origins";
+import {
+  Effect,
+  Group,
+  Policy,
+  PolicyStatement,
+  User,
+} from "aws-cdk-lib/aws-iam";
+import { ARecord, HostedZone, RecordTarget } from "aws-cdk-lib/aws-route53";
+import { CloudFrontTarget } from "aws-cdk-lib/aws-route53-targets";
+import {
+  BlockPublicAccess,
+  Bucket,
+  BucketEncryption,
+  BucketProps,
+} from "aws-cdk-lib/aws-s3";
 import { CSP } from "../types/csp";
-import { RecordTarget } from "@aws-cdk/aws-route53";
-import { CloudFrontTarget } from "@aws-cdk/aws-route53-targets";
 
 export interface StaticHostingProps {
+  domainNames: string[];
```

Comment:
Are we concerned about backwards compatibility? I think it's fine not to worry but we need a migration guide